modules/iap/iap.tf

```diff
@@ -0,0 +1,176 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# fetch all namespaces
+data "kubernetes_all_namespaces" "allns" {}
+
+resource "random_string" "random" {
+  length  = 4
+  special = false
+  upper   = false
+}
+
+# Jupyter IAP
+resource "google_compute_global_address" "jupyter_ip_address" {
+  count        = var.jupyter_add_auth && var.jupyter_url_domain_addr == "" ? 1 : 0
+  provider     = google-beta
+  project      = var.project_id
+  name         = "jupyter-address-${random_string.random.result}"
+  address_type = "EXTERNAL"
+  ip_version   = "IPV4"
+}
+
+# Helm Chart IAP
+resource "helm_release" "iap_jupyter" {
+  count            = var.jupyter_add_auth ? 1 : 0
+  name             = "iap-jupyter"
+  chart            = "${path.module}/charts/iap/"
+  namespace        = var.namespace
+  create_namespace = !contains(data.kubernetes_all_namespaces.allns.namespaces, var.namespace)
+  # timeout increased to support autopilot scaling resources, and give enough time to complete the deployment 
+  timeout = 1200
+  set {
+    name  = "iap.backendConfig.name"
+    value = var.jupyter_k8s_backend_config_name
+  }
+
+  set {
+    name  = "iap.secret.name"
+    value = var.jupyter_k8s_iap_secret_name
+  }
+
+  set {
+    name  = "iap.secret.client_id"
+    value = base64encode(var.jupyter_client_id)
+  }
+
+  set {
+    name  = "iap.secret.client_secret"
+    value = base64encode(var.jupyter_client_secret)
+  }
+
+  set {
+    name = "iap.managedCertificate.name"
+    value = var.jupyter_k8s_managed_cert_name
```

Comment:
remove all jupyter occurences from iap module..  jupyter keyword will appear in jupyter module only when calling iap module.

Reply:
Discussed offline, will refactor it in next PR